### ISSUE TITLE
Unable to upload new add-on version: "The submitted file is empty"

### ISSUE THREAD

**Issue by mcdavis941** [mozilla/addons#23]
I've tried several times but the upload fails with the message "The submitted file is empty".

(Also, the status changes to "validating" when the file is only around 40% uploaded.  The upload continues to 100%, and validation appears to complete, finding the expected warnings and giving a validation report, but that seems a little strange.  Maybe that's expected though.)

Screenshot: http://mcdavis.dreamhosters.com/imgshare/new-version-upload-error.20160125.png

**Comment by magopian** [comment]
Seems like a duplicate of https://github.com/mozilla/olympia/issues/1503
What Firefox version are you using, and what exactly do you do to have this issue? Do you click on the "submit" button several times? Have you tried waiting a bit before clicking on the submit button?

Is there somewhere I can download this new version you're trying to upload?

**Comment by mcdavis941** [comment]
> Have you tried waiting a bit before clicking on the submit button?

More or less, but that's a good idea, let me try again with a three minute wait in there.

Firefox version: the latest nightly, Mozilla/5.0 (Windows NT 6.0; WOW64; rv:47.0) Gecko/20100101 Firefox/47.0

Exact STR:
1.  Go to the Status and Versions page for the addon on AMO.
2.  Click the blue "Upload a new version" button (just under "Older Versions").
3.  In the modal box that overlays the page, click "Select a file..."
4.  From the system dialog box that appears, pick the addon file to upload, say OK.
5.  Note that "All platforms" is checked.
6.  The modal dialog shows uploading, with the progress bar and spinner.
7.  Somewhere between 13% and 50% the status message switches to "Validating" while the upload continues.
8  At 100% the message shows "Finished validating" with "Your add-on was validated with no errors and 9 warnings."
9. Click the "Add Version" button at the bottom to finish submitting.
10.  The message "The submitted file is empty." appears, and nothing else happens.

> Do you click on the "submit" button several times?

Good question but no, just once.

> Is there somewhere I can download this new version you're trying to upload?

Yes, I put it here, it's the same one:

http://mcdavis.dreamhosters.com/nnlopenbeta/nasanightlaunch_0.6.20160125.rc.jar

**Comment by magopian** [comment]
Can't reproduce locally, so I guess it's related to the production environment, I'll close this bug here and reopen the one on mozilla/olympia#1503, thanks for filing!

**Comment by mcdavis941** [comment]
> Have you tried waiting a bit before clicking on the submit button?

I tried again with a three minute wait and then again with a ten minute wait, and the result was the same, "The submitted file is empty."

**Comment by magopian** [comment]
Seems this is only happening for a Firefox version above 45 (so 46 and 47?)

**Comment by mcdavis941** [comment]
> Seems this is only happening for a Firefox version above 45 (so 46 and 47?)

That was a very good observation :) because it just worked successfully, as normal, with Firefox 43.0.4.

I'll bow out now, thanks for working the problem with me.

**Comment by julmot** [comment]
I have the same issue

**Comment by jvillalobos** [comment]
This issue was filed again as [bug 1249966](https://bugzilla.mozilla.org/show_bug.cgi?id=1249966). I'm using the latest Nightly, though, and I'm not seeing this error. Which version of Firefox are you using?

**Comment by julmot** [comment]
46.0a2 (2016-02-29)
